### DISCUSSION TITLE
Using `object_store` with a local S3-compatible endpoint

### DISCUSSION THREAD

**Discussion by Kinrany** [apache/arrow-rs#3867] [Q&A] [answered]
I'm using `object_store` with a [SeaweedFS](https://github.com/seaweedfs/seaweedfs) container, this way my whole development setup is local. I'm running into a problem where the client is trying and failing to connect to the AWS metadata endpoint, `http://169.254.169.254/latest/api/token`.

My first guess is that the client eagerly tries to get an authentication token before performing the S3 request. (Although the error is saying "Host is down", which seems to suggest some kind of routing error?)

Would it make sense for the client to try to authenticate lazily, after getting a 401 Unauthenticated or an equivalent response?

**Comment by tustvold** [comment, marked as answer]
The S3 implementation defaults to using InstanceMetadata based credentials for consistency with AWS SDKs. If you specify an alternative credential, e.g. an access key pair, it will use that instead

**Reply by Kinrany** [reply]
Hmm. I'm specifying an endpoint instead of a region and not providing any keys. Should I explicitly provide null credentials of some kind?
